google/cloud/bigquery_storage_v1beta2/client.py

```diff
@@ -123,19 +130,12 @@ def read_rows(
             ValueError: If the parameters are invalid.
         """
         gapic_client = super(BigQueryReadClient, self)
-        stream = gapic_client.read_rows(
```

Comment:
Note: A side-effect of this change is that some non-retryable errors won't happen until the user starts iterating through rows.

I would consider this a breaking change even more-so than the ReadRowsStream constructor change, as users who were expecting an exception now will get it later on.

Reply:
I changed the code so read_rows is called in ReadRowsStream construction and any exception besides the retryable resource exhausted is propagated.